core-rust-bridge/src/main/java/com/radixdlt/transaction/ExecutedTransaction.java

```diff
@@ -64,22 +64,19 @@
 
 package com.radixdlt.transaction;
 
-import com.radixdlt.rev2.ComponentAddress;
-import com.radixdlt.rev2.ResourceAddress;
 import com.radixdlt.sbor.codec.CodecMap;
 import com.radixdlt.sbor.codec.StructCodec;
 import com.radixdlt.transactions.RawLedgerTransaction;
+import com.radixdlt.utils.UInt64;
 import java.util.Arrays;
-import java.util.List;
 import java.util.Objects;
 
 /** A wrapper for a transaction and its ledger receipt */
 public record ExecutedTransaction(
+    UInt64 stateVersion,
```

Comment:
> it felt weird when you said "I added stateVersion to the result because it's needed".
I can still argue that a transaction after being executed (against a state) would give you a `stateVersion` (so it doesn't "rot" the definition). The query is done by state version because that's what the index gives you :).

> I also see that you already have a `getStateVersionForCommittedLedgerTransaction()` helper there
What that function does is iterate from `stateVersion` 1 until it finds the one which has the same transaction. I just couldn't let myself do that again for `TransactionDetails` :(.

> I typically argue that tests should not drive the API (and definitely should not make it uglier :p).
At least 100% onboard with this. But the damage has already been done (hence why the separation of `ExecutedTransaction` into what it is now and `TransactionDetails`). 

If you still think `stateVersion` doesn't have it's place there I can change it. I don't feel strongly about inefficient tests that much :)

Reply:
> I can still argue that a transaction after being executed (against a state) would give you a `stateVersion`

yes, but 🔍 

> a transaction after being executed

🔎 

> executed

I totally agree, that a method like `engine.execute(transaction_payload)` could return `new ExecutedTransaction(state_changes, whatever_else, state_version)`.
But a method like `store.get_executed_transaction_by_state_version(state_version)` totally does not have to echo-back the `state_version` to the caller.
In our case, we use `ExecutedTransaction` as a return type only from `get_executed_transaction_by_state_version()`.

If you can add a _simple_ (not necessarily _efficient_) test helper that lives without `state_version` - please do it.
If it is too cumbersome to test stuff without `state_version` - then ofc I can live with it.